oddt/surface.py

```diff
@@ -0,0 +1,128 @@
+"""This module generates, plots, and does computation with molecular surfaces.
+"""
+
+import numpy as np
+from skimage import measure
+from skimage.morphology import ball, binary_closing
+from scipy.spatial import cKDTree
+
+
+def generate_surface_marching_cubes(molecule, remove_hoh=False, scaling=1,
+                                    probe_radius=1.4):
+    """Generates a molecular surface mesh using the marching_cubes_lewiner
+    method from scikit-image.
+
+    Parameters
+    ----------
+    molecule : oddt.toolkit.Molecule object
+        Molecule for which the surface will be generated
+
+    remove_hoh : bool (default = False)
+        If True, remove waters from the molecule before generating the surface
+
+    scaling : float (default = 1)
+        Expands the grid in which computation is done by a factor of scaling.
+        Results in a more accurate representation of the surface, but increases
+        computation time.
+
+    probe_radius : float (default = 1.4)
+        Radius of a ball used to patch up holes inside the molecule
+        resulting from some molecular distances being larger
+        (usually in protein). Basically reduces the surface to one
+        accesible by other molecules of radius smaller than probe_radius.
+
+    Returns
+    -------
+    verts : numpy array
+        Spatial coordinates for mesh vertices.
+
+    faces : numpy array
+        Faces are defined by referencing vertices from verts.
+    """
+    atom_dict = molecule.atom_dict
```

Comment:
Add input validation please - we must be sure that each argument has a correct type and value (raise `TypeError` or `ValueError` with a meaningful message if input is incorrect).